app/models/supplier.rb

```diff
@@ -6,26 +6,35 @@ class Supplier < ActiveRecord::Base
   # save lists in an array in database
   serialize :lists
 
+  # @rails4: enum ftp_type: {bnn: 0, foodsoft: 1}
+  FTP_TYPES = {bnn: 0, foodsoft: 1}
+
   EMAIL_RE = /^([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})$/i.freeze
 
   validates :name, :address, :phone, presence: true
-  validates :bnn_host, :bnn_user, :bnn_password, :bnn_sync, presence: true, if: :bnn_sync?
+  validates :ftp_host, :ftp_user, :ftp_password, :ftp_sync, presence: true, if: :ftp_sync?
+  validates :ftp_type, presence: true, inclusion: {in: FTP_TYPES.values}, if: :ftp_sync?
```

Comment:
spaces?

Reply:
Added spaces in d664a26e99ac139785366d982b3d9b9de56faf5b .